tools/libraries/src/Server.ts

```diff
@@ -478,7 +520,8 @@ export class Server extends TypedEmitter<ServerEvents> {
       if (
         typeof flat === 'object' &&
         param in flat &&
-        flat[param].toLowerCase() !== params[param].toLowerCase()
+        String(flat[param]).toLowerCase() !==
```

Comment:
This cast on an object can end up with this representation : '[object Object]' for instance if flat[param] or params[param] is an object: [[]] or [{}] etc...

Reply:
maybe somehting like that ? 

  
  private compare(params: any, flat: any): Array<string> {
    const errors: Array<string> = []
    for (const param in params) {
      if (typeof flat === 'object' && param in flat ) {
        const flatP= flat[param]
        const paramP= params[param]
        if(typeof flatP != typeof paramP) {
          errors.push(param)
        }
        else if(typeof flatP == 'object') {
          if (compare(flatP, paramP)) {
            errors.push(param)
          }
        }
        else if(String(flatP).toLowerCase() !== String(paramP).toLowerCase()) {
          errors.push(param)
        }
      }else {
        errors.push(param)
      }
    }
    return errors
  }

Reply:
Request and response params for `get*OrderERC20` calls are shallow so these will only ever be stringish. Since it's only being used in those calls it should be okay.